runelite-client/src/main/java/net/runelite/client/ui/ClientUI.java

```diff
@@ -379,24 +385,28 @@ void expand(PluginPanel panel)
 	{
 		if (pluginPanel != null)
 		{
-			navContainer.remove(0);
+			activeNavContainer.remove(0);
 		}
 		else
 		{
-			clientSize = this.getSize();
 			if (isInScreenBounds((int) getLocationOnScreen().getX() + getWidth() + PANEL_EXPANDED_WIDTH, (int) getLocationOnScreen().getY()))
 			{
+				activeNavContainer = outerNavContainer;
 				this.setSize(getWidth() + PANEL_EXPANDED_WIDTH, getHeight());
 			}
+			else
+			{
+				activeNavContainer = innerNavContainer;
+			}
 		}
 
 		pluginPanel = panel;
-		navContainer.setMinimumSize(new Dimension(PANEL_EXPANDED_WIDTH, 0));
-		navContainer.setMaximumSize(new Dimension(PANEL_EXPANDED_WIDTH, Integer.MAX_VALUE));
+		activeNavContainer.setMinimumSize(new Dimension(PANEL_EXPANDED_WIDTH, 0));
+		activeNavContainer.setMaximumSize(new Dimension(PANEL_EXPANDED_WIDTH, Integer.MAX_VALUE));
```

Comment:
why not just always keep the maximum size at the PANEL_EXPANDED_WIDTH one but change it every time?

Reply:
It works just fine when setting the maximum size to `PANEL_EXPANDED_WIDTH` when creating `outerNavContainer` and `innerNavContainer` so I guess I should just do that.